view/src/main/java/dev/marcellogalhardo/retained/view/ViewRetainedObject.kt

```diff
@@ -0,0 +1,52 @@
+package dev.marcellogalhardo.retained.view
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.navigation.NavBackStackEntry
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import dev.marcellogalhardo.retained.core.InternalRetainedApi
+import dev.marcellogalhardo.retained.core.RetainedEntry
+import dev.marcellogalhardo.retained.core.createRetainedObjectLazy
+
+/**
+ * Returns a [Lazy] delegate to access a retained object by **default** scoped to this
+ * [View] (e.g., [NavBackStackEntry], [Fragment] or [Activity]).
+ *
+ * ```
+ * class MyView(context: Context) : View(context) {
+ *     val vm by retain { ViewModel() }
+ * }
+ * class ViewModel(val name: String = "")
+ * ```
+ *
+ * This property can be accessed only after the [LifecycleOwner] is ready to use Jetpack
+ * [ViewModel], and access prior to that will result in [IllegalArgumentException]. TODO: right now it actually throws a NPE
+ *
+ * @see createRetainedObject
+ */
+@OptIn(InternalRetainedApi::class)
+inline fun <reified T : Any> View.retain(
+    key: String = T::class.java.name,
+    noinline getViewModelStoreOwner: () -> ViewModelStoreOwner = { findViewTreeViewModelStoreOwner()!! },
```

Comment:
I'm not sure if it is a good idea. Still, after retrieving the owners, we might want to execute a `ViewTreeViewModelStoreOwner.set(view, viewModelStoreOwner)` to ensure a child view would not need to go upper from this view and reduce the need to transverse the view tree. Same would apply for the other `owner` objects...

Reply:
Will resolve this for now and we can get back to it in a following iteration.